float32 x_min
float32 y_min
float32 x_max
float32 y_max
float32 center_x
float32 center_y